src/Microsoft.DncEng.SecretManager/SecretTypes/TextSecretType.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DncEng.CommandLineLib;
+
+namespace Microsoft.DncEng.SecretManager.SecretTypes
+{
+    [Name("text")]
+    public class TextSecretType : SecretType<TextSecretType.Parameters>
+    {
+        private readonly IConsole _console;
+
+        public class Parameters
+        {
+            public string Description { get; set; }
+        }
+
+        public TextSecretType(IConsole console)
+        {
+            _console = console;
+        }
+
+        protected override async Task<SecretData> RotateValue(Parameters parameters, RotationContext context, CancellationToken cancellationToken)
+        {
+            var existing = await context.GetSecretValue(context.SecretName);
+            if (!_console.IsInteractive)
+            {
+                throw new InvalidOperationException($"Text secret doesn't exist. Human intervention required.");
+            }
+            var newValue = await _console.PromptAsync($"Input value for {context.SecretName} (empty to keep existing), {parameters.Description}: ");
+            if (string.IsNullOrEmpty(newValue))
```

Comment:
What's the scenario that you want to do nothing?  Should we log a warning of something like "I didn't do anything"?

Reply:
I could make it error, but these "secrets" that aren't actually secrets almost never change. Having a way to say "don't actually rotate this" is a good thing I think. The value will be something like the github app id, so it actually can't ever change.

Reply:
Ah. These are the "these really shouldn't be in keyvault, but we used it as a settings store" type secrets?

Reply:
Hopefully we don't have many of those left, and even extra hopefully, we can move the ones we find to more appropriate places as part of this effort if we have to touch a bunch of stuff anyway.